blocks/api/registration.js

```diff
@@ -125,6 +125,17 @@ export function registerBlockType( name, settings ) {
 		...settings,
 	};
 
+	// Resolve deprecated attributes
+	settings.attributes = mapValues( settings.attributes, ( attribute ) => {
+		if ( isFunction( attribute.source ) ) {
```

Comment:
Do we need to return functions from the deprecated matcher? Why not just return the object, and skip this step?

Reply:
Because it won't match the current API.

it will produce: `source: { source: 'text' }` which is not the correct attribute definition

Reply:
This could also be implemented as a filter hook, which has the added advantage of being simpler to drop 😄

Reply:
Great idea